src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ClassImportControl.java

```diff
@@ -0,0 +1,99 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.imports;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents a tree of import rules for a specific class.
```

Comment:
please make it clear that it is not for inner/nested classes. It is only for top level public classes ?
please update xdoc too.

What about Interfaces, Enums (all other top level types) ?  
It might be another issue .... but should we change DTD in such a way to allow extending logic without DTD changes. So in stead of "Class" term use "Type".

Interfaces imports are even more critical target of such Import Control validation, as such imports force clients of API have all that classes/types as compile dependency.

Reply:
> but should we change DTD in such a way to allow extending logic without DTD changes. 

For me, right now I would say no. Having the type in the tag name really distinguishes it from others around it and it helps shrink the size of the import control file. Mine has 620 lines because I build a whitelist for all packages.

> It is only for top level public classes ?
> What about Interfaces, Enums (all other top level types) ?

yes to both. Top level only and allows all types of classes (interface, enum, etc...). You can see the tests for them.
I will update the documentation.

Reply:
Done.